public/app/plugins/panel/geomap/layers/data/heatMap.tsx

```diff
@@ -77,47 +83,55 @@ export const heatmapLayer: MapLayerRegistryItem<HeatmapConfig> = {
           return; // ???
         }
 
-        // Get the field of data values
-        const field = frame.fields.find(field => field.type === FieldType.number); // TODO!!!!
-        // Return early if metric field is not matched
-        if (field === undefined) {
-          return;
-        };
+        const weightDim = getScaledDimension(frame, config.weight);
 
-        // Retrieve the min, max and range of data values
-        const calcs = reduceField({
-          field: field,
-          reducers: [
-            ReducerID.min,
-            ReducerID.range,
-          ]
-        });
         // Map each data value into new points
         for (let i = 0; i < frame.length; i++) {
           const cluster = new Feature({
               geometry: info.points[i],
-              value: normalize(calcs, field.values.get(i)),
+              value: weightDim.get(i),
           });
           vectorSource.addFeature(cluster);
         };
         vectorLayer.setSource(vectorSource);
 
-        // Set gradient of heatmap
-        const colorMode = getFieldColorModeForField(field);
-        if (colorMode.isContinuous && colorMode.getColors) {
-          // getColors return an array of color string from the color scheme chosen
-          const colors = colorMode.getColors(theme);
-          vectorLayer.setGradient(colors);
-        } else {
-          // Set the gradient back to default if threshold or single color is chosen
-          vectorLayer.setGradient(['#00f', '#0ff', '#0f0', '#ff0', '#f00']);
+        // Set heatmap gradient colors
+        let colors = ['#00f', '#0ff', '#0f0', '#ff0', '#f00'];
+        let field = weightDim.field;
+        if(!field) { // Use the first nummeric field color scheme
+          field = frame.fields.find(field => field.type === FieldType.number);
         }
+        if(field) {
```

Comment:
else?

Reply:
:grimacing: ya that looks pretty weird.  I updated the logic to:
```
        const field = weightDim.field ?? frame.fields.find(field => field.type === FieldType.number);
```